api/v1beta1/glance_webhook.go

```diff
@@ -23,19 +23,51 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
+// GlanceDefaults -
+type GlanceDefaults struct {
+	ContainerImageURL string
+}
+
+var glanceDefaults GlanceDefaults
+
 // log is for logging in this package.
 var glancelog = logf.Log.WithName("glance-resource")
 
 // SetupWebhookWithManager sets up the webhook with the Manager
-func (r *Glance) SetupWebhookWithManager(mgr ctrl.Manager) error {
+func (r *Glance) SetupWebhookWithManager(mgr ctrl.Manager, defaults GlanceDefaults) error {
+	glanceDefaults = defaults
+
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
 		Complete()
 }
 
-// TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+//+kubebuilder:webhook:path=/mutate-glance-openstack-org-v1beta1-glance,mutating=true,failurePolicy=fail,sideEffects=None,groups=glance.openstack.org,resources=glances,verbs=create;update,versions=v1beta1,name=mglance.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &Glance{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *Glance) Default() {
+	glancelog.Info("default", "name", r.Name)
+
+	r.Spec.Default()
+}
+
+// Default - set defaults for this Glance spec
+func (spec *GlanceSpec) Default() {
```

Comment:
@gibizer this is reusable because it's part of the `api/` and  can be imported in the `openstack-operator`, right? 
By doing this we can enforce defaults from  `meta-operator` (populating `GlanceDefaults`, which is passed to `SetupWebhookWithManager`), then, when the service operator is executed and the webhook is started, we can process those default values via this function (where the logic resides).
Is that the right understanding of the potential workflow?